libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -0,0 +1,354 @@
+    case 'passed':
+      return ExecJSON.ControlResultStatus.Passed;
+    case 'open':
+    case 'failed':
+      return ExecJSON.ControlResultStatus.Failed;
+    case 'error':
+      return ExecJSON.ControlResultStatus.Error;
+    default:
+      return ExecJSON.ControlResultStatus.Skipped;
+  }
+}
+
+/**
+ * Transformer function that uses current heimdall checklist export syntax for
+ * findingDetails attribute to separate a single string into multiple
+ * result objects
+ * @param input - array of one element consisting of {code_desc, status, start_time}
+ * @returns ExecJSON.ControlResult
+ */
+function parseFindingDetails(input: unknown[]): ExecJSON.ControlResult[] {
+  const findings = input as unknown as ExecJSON.ControlResult[];
+  const results: ExecJSON.ControlResult[] = [];
+  const statusSet = ['passed', 'failed', 'skipped', 'error'];
+
+  for (const finding of findings) {
+    if (finding.code_desc) {
+      // split into multiple findings details using heimdall2 CKLExport functionality
+      for (const details of finding.code_desc.split(
+        '--------------------------------\n'
+      )) {
+        let code_desc: string;
+        let status: ExecJSON.ControlResultStatus;
+        let message = '';
+        // split details for status
+        const [findingStatus, descAndMessage] = details.split(/\n(.*)/s, 2);
+        if (statusSet.includes(findingStatus)) {
+          // split desc and message
+          // This does not necessarily work when the code_desc has the word expected
+          // will need to update the export to add a key word that can be used to split code_desc
+          // and message values easier - maybe *results* or *details*
+          const indexOfExpected = descAndMessage.indexOf('\nexpected');
+          if (indexOfExpected > 0) {
+            code_desc = descAndMessage.slice(0, indexOfExpected - 1);
+            message = descAndMessage.slice(indexOfExpected);
+            status = getStatus(findingStatus);
+          } else {
+            code_desc = descAndMessage;
+            status = getStatus(findingStatus);
+          }
+        } else {
+          code_desc = details;
+          status = finding.status as ExecJSON.ControlResultStatus;
+        }
+        results.push({
+          code_desc,
+          status,
+          message: message ? message : null,
+          start_time: ''
+        });
+      }
+    } else {
+      results.push(finding);
+    }
+  }
+  return results;
+}
+
+/**
+ * ChecklistResults is a wrapper for ChecklistMapper using the intakeType
+ *  default returns a single hdf object without any modifications
+ *  split returns multiple hdf object based on number of iSTIG objects in checklist
+ *  wrapper returns a single hdf object with an additional profile created using file name as profile name and adds parent_profile key to each mapped profile
+ */
+export class ChecklistResults extends ChecklistJsonixConverter {
+  checklistXml: string;
+  raw: ChecklistJSONIX;
```

Comment:
raw should be reserved for the original xml string, please change names accordingly.  ex. here instead of raw for the checklistjsonix type, maybe we can use jsonix_firstpass or jsonix_raw or whatever idk